### PR TITLE
Remove prometheus addon

### PR DESCRIPTION
**Pull request by serathius**
/kind cleanup

```release-note
Remove prometheus cluster monitoring addon from kube-up
```
/cc @kawych

### CODE REVIEW

**Comment by serathius** [comment]
/cc @bentheelder 
/cc @MaciekPytel 
PTAL

**Comment by BenTheElder** [comment]
> Addon was never meant to be used by users, not need to add release note.

Hmm not meant to != Was not, was this bundled in releases?

**Comment by serathius** [comment]
It was not meant to. This addon was bundled in kubeup, so I can be deployed for periodic tests https://k8s-testgrid.appspot.com/sig-instrumentation#gce-prometheus
I looked on PR adding it https://github.com/kubernetes/kubernetes/pull/62195
It was mentioned in release notes, so we should mention it deletion.

**Comment by serathius** [comment]
ping @BenTheElder

File: test/e2e/framework/skip.go [hunk 1]
```diff
@@ -108,13 +108,6 @@ func SkipIfMultizone(c clientset.Interface) {
 	}
 }
 
-// SkipUnlessPrometheusMonitoringIsEnabled skips if the prometheus monitoring is not enabled.
-func SkipUnlessPrometheusMonitoringIsEnabled(supportedMonitoring ...string) {
-	if !TestContext.EnablePrometheusMonitoring {
-		skipInternalf(1, "Skipped because prometheus monitoring is not enabled")
-	}
-}
```
Comment (alejandrox1) [inline test/e2e/framework/skip.go]: This PR removes the following from the test context as well:
```go
// Separate Prometheus monitoring deployed in cluster
EnablePrometheusMonitoring bool
```

As far as I know, the e2e test framework is not (should not) be tied to anything in the cluster directory.
I think in general, we should keep this bit.

File: test/e2e/framework/test_context.go [hunk 2]
```diff
@@ -151,8 +151,6 @@ type TestContextType struct {
 	NodeTestContextType
 	// Monitoring solution that is used in current cluster.
 	ClusterMonitoringMode string
-	// Separate Prometheus monitoring deployed in cluster
-	EnablePrometheusMonitoring bool
```
Comment (alejandrox1) [inline test/e2e/framework/test_context.go]: [1]

File: test/e2e/framework/test_context.go [hunk 3]
```diff
@@ -313,7 +311,6 @@ func RegisterClusterFlags(flags *flag.FlagSet) {
 	flags.StringVar(&TestContext.MasterOSDistro, "master-os-distro", "debian", "The OS distribution of cluster master (debian, ubuntu, gci, coreos, or custom).")
 	flags.StringVar(&TestContext.NodeOSDistro, "node-os-distro", "debian", "The OS distribution of cluster VM instances (debian, ubuntu, gci, coreos, or custom).")
 	flags.StringVar(&TestContext.ClusterMonitoringMode, "cluster-monitoring-mode", "standalone", "The monitoring solution that is used in the cluster.")
-	flags.BoolVar(&TestContext.EnablePrometheusMonitoring, "prometheus-monitoring", false, "Separate Prometheus monitoring deployed in cluster.")
```
Comment (alejandrox1) [inline test/e2e/framework/test_context.go]: [1]

File: test/e2e/instrumentation/monitoring/prometheus.go [hunk 4]
```diff
@@ -1,388 +0,0 @@
...
-	prometheusRate                     = time.Minute * 2
-	prometheusRequiredNodesUpDuration  = time.Minute * 5
-	prometheusService                  = "prometheus"
-	prometheusSleepBetweenAttempts     = time.Second * 30
-	prometheusTestTimeout              = time.Minute * 5
-	customMetricValue                  = 1000
-	targetCPUUsage                     = 0.1
-)
-
-var _ = instrumentation.SIGDescribe("[Feature:PrometheusMonitoring] Prometheus", func() {
```
Comment (alejandrox1) [inline test/e2e/instrumentation/monitoring/prometheus.go]: Do all of these tests depend strictly on the cluster addon from the `cluster/`?
If prometheus was to be enabled using some method other than kube-up, should this be still removed?

**Review by alejandrox1 — COMMENTED** [review]
/hold
/assign @timothysc @neolit123 @andrewsykim 
ptal

**Comment by alejandrox1** [comment]
quick question: whats the reason for removing the prometheus addon from kube-up?

/priority important-soon
/milestone v1.17

File: test/e2e/framework/skip.go (hunk 1 above)
Comment (BenTheElder) [reply test/e2e/framework/skip.go]: > As far as I know, the e2e test framework is not [...] be tied to anything in the cluster directory.

That's wildly untrue? : - )
The e2e test is tied to EG `log-dump.sh`, and more ...

> As far as I know, the e2e test framework [...] (should not) be tied to anything in the cluster directory.

That's debateable. I don't think we have a definition anywhere of what it should be tied to. It's also tied to lots of cloud providers..

Comment (BenTheElder) [reply test/e2e/framework/skip.go]: (no comment on this particular bool being removed, but I don't think those assertions are accurate)

Comment (alejandrox1) [reply test/e2e/framework/skip.go]: mmm this is cool. So I guess the implementation of some set of features of the e2e test framework may be tied to the way things are configured in the cluster directory.
And maybe this part of the framework needs to be reworked to be more general (?). 

Ah so this is cool. I was wondering if the bits that are being removed would impact the rest of the framework in any way. and going through the code and it seems that the things that are being removed are really only used in the instrumentation tests that are being deleted, so I guess this doesn't affect any other framework functionality.

Made my initial comment thinking that even though the framework may have some bits that assume things were setup using code from the cluster directory, what lives in the cluster directory should not influence the test framework. i.e., we should make things in the framework more general; make it easier for people to consume...
/hold cancel

**Comment by alejandrox1** [comment]
Changes to the e2etest framework don't seem to affect the framework in general so gonna undue my comments here.
/unassign @timothysc @andrewsykim

File: test/e2e/framework/skip.go (hunk 1 above)
Comment (oomichi) [reply test/e2e/framework/skip.go]: According to https://cs.k8s.io/?q=EnablePrometheusMonitoring&i=nope&files=&repos= EnablePrometheusMonitoring is used at here only and the function SkipUnlessPrometheusMonitoringIsEnabled() is called at `test/e2e/instrumentation/monitoring/prometheus.go` which is removed in this PR as https://cs.k8s.io/?q=SkipUnlessPrometheusMonitoringIsEnabled&i=nope&files=&repos=

So it seems fine to remove this part in this PR for me.

**Review by oomichi — COMMENTED** [review]
For e2e framework

/approve

File: test/e2e/instrumentation/monitoring/prometheus.go (hunk 4 above)
Comment (serathius) [reply test/e2e/instrumentation/monitoring/prometheus.go]: Those tests where strictly developed for cluster addon in `cluster`. There is no officially supported  prometheus distribution by kubernetes community.

**Comment by serathius** [comment]
Main reason for removing it is maintenance cost for addon that outweigh benefits of testing generic prometheus configuration. There are much more advanced prometheus deployments manifests that take different approach to configuring prometheus that doesn't benefit from tests in k/k repo.

**Comment by cblecker** [comment]
/approve
for hack/

**Comment by markyjackson-taulia** [comment]
Bug triage for 1.17 here with a gentle reminder that code freeze for this release is on November 18. Is this issue still intended for 1.17?

**Comment by serathius** [comment]
There is no target k8s version that this change is targeting.

**Review by timothysc — APPROVED** [review]
/approve 
/assign @timothysc

**Pull request merged**